packages/v4/patternfly-docs/content/accessibility/menu/menu.md

```diff
@@ -0,0 +1,53 @@
+---
+id: Menu
+section: components
+---
+
+import { Checkbox, List, ListItem } from '@patternfly/react-core';
+
+
+## Accessibility
+
+To implement an accessible PatternFly **menu** component:
+- Due to the composable nature of the menu component, keyboard interaction will need to be added manually between the menu toggle and the menu. See the [composable menu demos](https://www.patternfly.org/v4/demos/composable-menu) to see how this is done.
+- Any actions placed within a menu item will need an `aria-label` on the `MenuItemAction` to communicate to users what the action is.
+
+
+## Testing
+
+At a minimum, a menu component should meet the following criteria:
+
+<List isPlain>
+  <ListItem>
+    <Checkbox id="menu-a11y-checkbox-1" label={<span>Users should be able to use keyboard navigation to enter the menu, navigate between menu items, select an option, or escape the menu.</span>} description={<span><kbd>Tab</kbd> navigates to the menu, <kbd>Shift</kbd> + <kbd>Tab</kbd> navigates to the previous focusable element, up and down arrow keys should be used to navigate between menu times, and <kbd>Escape</kbd> should close the menu."</span>}/>
+  </ListItem>
+  <ListItem>
+    <Checkbox id="menu-a11y-checkbox-2" label={<span>If using menu item actions, the actions have an accessible name via the <code className="ws-code">aria-label</code></span>} description="This will differentiate each action when announced to users of assistive technologies when navigating through a page or a screen reader's rotor menu." />
+  </ListItem>
+</List>
+
+
+## React customization
+
+Various React props have been provided for more fine-tuned control over accessibility.
+
+| Prop | Applied to | Reason | 
+|---|---|---|
+| `aria-label` | `Menu` | Adds an accessible name to the menu. |
```

Comment:
Adding an `aria-label` to the Menu could make sense in some situations, but I'm not sure I'd consider it required? We don't have it on any of our examples/demos. Do you think it's something we should be adding?

Reply:
Looking into it some more, an `aria-label` on the Menu probably wouldn't be required. An `aria-label` on the MenuList sub-component might be worth requiring since placing it there would help differentiate the menu in a screen reader rotor.

That would mean updating some examples + React code, possibly as a breaking change since it might make more sense to remove the aria-label from the Menu and just move it to MenuList. I'd be good with keeping this as-is for now, and it can be revisited if we want to make these updates in React and examples.